src/server.cpp

```diff
@@ -48,11 +56,35 @@ struct source_t {
 
 source_t sources[max_sources];
 
+// Create the database object
+CppSQLite3DB db;
+
+void db_table(){
+    db.execDML("create table if not exists pi(pk_pi integer primary key autoincrement, name char(20) unique);");
+    db.execDML("create table if not exists source(pk_source integer primary key autoincrement, fk_pi integer, foreign key(fk_pi) references pi(pk_pi));");
+    db.execDML("create table if not exists sensor(pk_sensor integer primary key autoincrement, type char(20), name char(20), fk_source integer, foreign key(fk_source) references source(pk_source));");
+    db.execDML("create table if not exists actuator(pk_actuator integer primary key autoincrement, name char(20), fk_source integer, foreign key(fk_source) references source(pk_source));");
+    db.execDML("create table if not exists sensor_data(pk_sensor_data integer primary key autoincrement, data char(20), time datetime not null default current_timestamp, fk_sensor integer, foreign key(fk_sensor) references sensor(pk_sensor));");
+    db.execDML("create table if not exists actuator_data(pk_actuator_data integer primary key autoincrement, data char(20), time datetime not null default current_timestamp, fk_actuator integer, foreign key(fk_actuator) references actuator(pk_actuator));");
+}
+
+void db_insert(){
+    db.execDML("insert into pi(name) select 'tyr' where not exists(select 1 from pi where name='tyr');");
+    db.execDML("insert into source(fk_pi) select 1 where not exists(select 1 from source where fk_pi=1);");
+    db.execDML("insert into sensor(type, name, fk_source) select 'TEMPERATURE', 'Local', 1 where not exists(select 1 from sensor where type='TEMPERATURE' and name='Local' and fk_source=1);");
```

Comment:
Most of these commands should be done when registering sources and actuators/sensors.